.github/workflows/ci.yml

```diff
@@ -194,6 +201,111 @@ jobs:
           if-no-files-found: error
           retention-days: 14
 
+  Test-Suite-Fork:
+    needs: [Precheck]
+    environment: Testing
+    if: ((startsWith(github.ref, 'refs/tags/') ||
+      github.ref == 'refs/heads/main' ||
+      needs.precheck.outputs.srcfileschanged == 'true') &&
+      github.event.pull_request.head.repo.full_name == github.repository) != true # skip when trusted workflow runs
```

Comment:
@meain let me know if you can think of a cleaner way to check this 🤔

Reply:
I think we can just check `needs.precheck.outputs.srcfileschanged == 'true' && github.event.pull_request.head.repo.full_name != github.repository` as we don't have to run this for `main` or tags.